docs/pipeline.md

```diff
@@ -0,0 +1,177 @@
+
+* Not be an instance of a [***blacklisted class***](../config/default.yml#L1-L9).
+* Have at least one [***unreferenced***](../config/default.yml#L108-L110) statement of a [***non blacklisted property***](../config/default.yml#L11-L76).
+* Have at least one External Id linking to a ***whitelisted*** external resource (see: [SS 1: External Resource Whitelister](#ss-1-external-resource-whitelister)).
+
+It takes in a de-serialized [Wikibase Item](https://gerrit.wikimedia.org/r/plugins/gitiles/mediawiki/extensions/Wikibase/+/master/docs/topics/json.md#json) dump as an input and returns an array with a single [`ItemLine`](result.md#itemline) or an empty array if the item does not match the criteria.
+
+### Pipe 2: Scraper
+
+[[Code]](../wikidatarefisland/pipes/scraper.py), [[Makefile Command]](../Makefile#L14-L17): `make data/scraped_data.jsonl`
+
+The scraper pipe makes calls to external resource URLs to retrieve Schema.org data in various formats (microdata, json-ld, rdfa). It [normalizes](#schemaorg-data-normalizer) this data and pairs statements with potential extracted data matches according to a Schema.org <-> Wikidata [mapping](#wikidata---schemaorg-property-mapper) maintained on Wikidata by the community.
+
+This pipe takes in a single [`ItemLine`](result.md#itemline) and returns an array with 0 or more [`MatchLine`](result.md#matchline)s, depending whether any potential matches were found.
+
+### Pipe 3: Value Matcher
+
+[[Code]](../wikidatarefisland/pipes/value_matcher_pipe.py), [[Makefile Command]](../Makefile#L18-L20): `make data/matched_references.jsonl`
+
+The value matcher pipe tries to make [exact matches](#wikibase-value-matchers) between the statement data and the extracted data for that statement.
+
+This pipe takes in a single [`MatchLine`](result.md#matchline), and returns an array with that line if a match exists or an empty array if it did not find a match.
+
+### Pipe 4: Statistical Matcher
+
+[[Code: Statistical Analysis]](../wikidatarefisland/pipes/item_statistical_analysis_pipe.py), [[Code: Item Matching]](../wikidatarefisland/pipes/item_mapping_matcher_pipe.py), [[Makefile Command]](../Makefile#L21-L23): `make data/matched_item_references.jsonl`
+
+This pipeline segment actually consist of two pipe steps in order to match extracted data and statement values which refer to a Wikibase Item: 
+
+- The first step reads and updates an internal statistic record about the frequency of matches between a piece of extracted data and a Wikibase QID utilizing settings for a [minimum frequency](../config/default.yml#L128-L129) and maximum amount of [allowed noise](../config/default.yml#L131-L132).
+-  The second step of this segment relies on the data gathered above to match Wikibase Items to their ***probable matches***.
+
+This segment takes in a single [`MatchLine`](result.md#matchline), and returns an array with that line if a match exists, or an empty array if it doesn't.
```

Comment:
It might be worth mentioning that this pipe actually reads the input data twice

Reply:
Will do!